verification/src/transaction_verifier.rs

```diff
@@ -124,6 +137,41 @@ impl<'a> EmptyVerifier<'a> {
     }
 }
 
+pub struct MaturityVerifier<'a> {
+    transaction: &'a Transaction,
+}
+
+impl<'a> MaturityVerifier<'a> {
+    pub fn new(transaction: &'a Transaction) -> Self {
+        Self { transaction }
+    }
+
+    pub fn verify(
+        &self,
+        block_number: BlockNumber,
+        cell_set: &FnvHashMap<H256, TransactionMeta>,
+        cellbase_maturity: usize,
+    ) -> Result<(), TransactionError> {
+        let immature_spend = self.transaction.inputs().iter().any(|input| {
+            match cell_set.get(&input.previous_output.hash) {
+                Some(ref meta)
+                    if meta.is_cellbase()
```

Comment:
Not every cell outpoint in cellbase need to be maturity, I think we need more discussion on this?

Consider the UDT tx fee design, maybe we should leave more rooms for upper UDT contracts?

Reply:
> Not every cell outpoint in cellbase need to be maturity, I think we need more discussion on this?
> 
> Consider the UDT tx fee design, maybe we should leave more rooms for upper UDT contracts?

I think it should be because of the block confirmation. The UDT tx fee would be included in the cellbase?

Reply:
Not decide yet.

Reply:
Seem your implement is correct due to the explanation of maturity in this link https://bitcoin.stackexchange.com/questions/1991/what-is-the-block-maturation-time

Anyway please hold this PR until #372 merged, 372 introduce few refactoring about tx verification.